torch/lib/THCUNN/LookupTable.cu

```diff
@@ -162,36 +162,76 @@ __global__ void cunn_LookupTable_accGradParametersKernel(
   }
 }
 
-/*
- * Keep the norm of weight smaller than maxNorm
- */
 template <typename Dtype, typename Acctype>
-struct pow_v
-{
-  Acctype normType;
-  pow_v(Dtype v) : normType(ScalarConvert<Dtype, Acctype>::to(v)) {}
-  __host__ __device__
-  Acctype operator()(const Dtype& x) const {
+__device__
+Acctype fast_pow(Dtype x, Acctype normType) {
     Acctype xA = ScalarConvert<Dtype, Acctype>::to(x);
     if (normType == 1)
       return std::abs(xA);
     else if (normType == 2)
       return xA * xA;
     else
       return std::pow(std::abs(xA), normType);
-  }
-};
+}
+
+template <typename Dtype>
+__device__
+Dtype get_padded_value(Dtype *weight_ptr, THCIndex_t idx, int64_t dim) 
+{
+  if (idx < dim)
+    return weight_ptr[idx];
+  else
+    return 0;
+}
 
-template <typename T>
-struct multiply_s
+/* Calculate norms of the rows of weight_ptr given by idx_ptr and capture them in norms */
+template <typename Dtype, typename Acctype>
+__global__
+void calculate_norms_and_renorm(Dtype *weight_ptr, THCIndex_t *idx_ptr, Dtype normType, 
+    Dtype maxNorm, Acctype normType_, int64_t dim)
```

Comment:
nit: can you clean up the argument names while you are at it? Remove the trailing underscore and keep the camelCase, and convert the other names to camelCase. The `ptr` part is a little excessive as well.

Also, unless all the parameters fit on a single line, can you put one on each line, aligned to the first parameter?